src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -117,22 +138,29 @@ public bool IsEmpty
         /// <param name="work">The WorkItem to process</param>
         public void Enqueue(WorkItem work)
         {
-            lock (_syncRoot)
+            do
```

Comment:
Can you explain your reasoning behind the changes here and in Dequeue? What is wrong with protecting the queue/dequeue with a lock and why is your spinning approach better?